utils/result_try.hh

```diff
@@ -373,7 +373,7 @@ public:
 
     auto with_original_result(R& res) {
         return [this, &res] (const auto& ex) {
-            return handle(ex, res);
+            return this->handle(ex, res);
```

Comment:
It's supposed to be illegal to call handle(...) without this->, because handle() is a dependent name (but many compilers don't insist, gcc is stricter here). So two error messages competed, and "unused this capture" won.

Anyway, the fix is good.

Reply:
@avikivity ahh, right! `R` is a template argument. i corrected the commit message by quoting your comment.

Reply:
FWIW, i am using Clang-17 + libstdc++ from GCC 13 here. so the warnings came from Clang.